pandas/io/sql.py

```diff
@@ -41,6 +41,22 @@ class DatabaseError(IOError):
 _SQLALCHEMY_INSTALLED = None
 
 
+def _validate_flavor_parameter(flavor):
+    """
+    Checks whether a database 'flavor' was specified.
+    If not None, produces FutureWarning if 'sqlite' and
+    raises a ValueError if anything else.
+    """
+    if flavor is not None:
+        if flavor == 'sqlite':
+            warnings.warn("the 'flavor' parameter is deprecated "
+                          "and will be removed in a future version.",
```

Comment:
and the same here. Something like: "it is not needed to specify this, because 'sqlite' is the default and only supported flavor when not using sqlalchemy" (so users don't think using sqlite itself is deprecated)